File: CS/ConsoleApplication/Model1.cs
using DevExpress.Data.Filtering;
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ConsoleApplication {

    public class Contact : XPLiteObject {
        int fContactID;
        [Key(true)]
        public int ContactID {
            get { return fContactID; }
            set { SetPropertyValue("ContactID", ref fContactID, value); }
        }
        string fFirstName;
        public string FirstName {
            get { return fFirstName; }
            set { SetPropertyValue("FirstName", ref fFirstName, value); }
        }
        string fLastName;
        public string LastName {
            get { return fLastName; }
            set { SetPropertyValue("LastName", ref fLastName, value); }
        }

        public XPCollection<Task> GetTasks(Session foreignSession) {
            return new XPCollection<Task>(foreignSession, new BinaryOperator("ContactID", this.ContactID));
        }

        public Contact(Session session) : base(session) { }
    }

}

File: VB/ConsoleApplication/Model1.vb
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace ConsoleApplication

    Public Class Contact
        Inherits XPLiteObject

        Private fContactID As Integer
        <Key(True)> _
        Public Property ContactID() As Integer
            Get
                Return fContactID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue("ContactID", fContactID, value)
            End Set
        End Property
        Private fFirstName As String
        Public Property FirstName() As String
            Get
                Return fFirstName
            End Get
            Set(ByVal value As String)
                SetPropertyValue("FirstName", fFirstName, value)
            End Set
        End Property
        Private fLastName As String
        Public Property LastName() As String
            Get
                Return fLastName
            End Get
            Set(ByVal value As String)
                SetPropertyValue("LastName", fLastName, value)
            End Set
        End Property

        Public Function GetTasks(ByVal foreignSession As Session) As XPCollection(Of Task)
            Return New XPCollection(Of Task)(foreignSession, New BinaryOperator("ContactID", Me.ContactID))
        End Function

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
    End Class

End Namespace
